src/inmanta/deploy.py

```diff
@@ -208,16 +209,16 @@ def _create_environment(self, project_id: str, environment_name: str) -> Optiona
 
         return env_id
 
-    def _latest_version(self, environment_id: str) -> Optional[int]:
-        result = self._client.list_versions(tid=environment_id)
+    def _latest_version_instance(self, environment_id: str) -> Optional[JsonType]:
+        result: Result = self._client.list_versions(tid=environment_id)
         if result.code != 200:
             LOGGER.error("Unable to get all version of environment %s", environment_id)
             return None
 
-        if "versions" in result.result and len(result.result["versions"]) > 0:
-            versions: List[int] = [x["version"] for x in result.result["versions"]]
-            sorted(versions)
-            return versions[0]
+        if result.result and "versions" in result.result and len(result.result["versions"]) > 0:
+            version = result.result["versions"][0]
+            assert isinstance(version, dict)  # mypy
+            return version
```

Comment:
You dropped the sorting but `list_versions` doesn't make any guarantees about the ordering (though perhaps it should).

Reply:
the sorting didn't do anything, as 'sorted' doesn't do an in place sort, but returns a sorted list, that was never assigned to anything.

Reply:
Ah, I didn't notice the lack of assignment